From 0b236dfd11f6f64a28037584b476398ba9abc880 Mon Sep 17 00:00:00 2001
From: Tsuyoshi Ozawa <ozawa@apache.org>
Date: Tue, 14 Jul 2015 19:13:45 +0900
Subject: [PATCH 1190/2748] HADOOP-12191. Bzip2Factory is not thread safe.
 Contributed by Brahma Reddy Battula.

(cherry picked from commit 4084eaf94347042d9b8cb1e86ba831036621dfdd)
(cherry picked from commit 0744a578e41ff28bd33088ae4e16f362d69a00a5)

Change-Id: I44fe8aed5d19ca7fc6aafa39bb9817140c7e4847
---
 .../hadoop/io/compress/bzip2/Bzip2Factory.java     |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/bzip2/Bzip2Factory.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/bzip2/Bzip2Factory.java
index c82ac99..0bbcc36 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/bzip2/Bzip2Factory.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/bzip2/Bzip2Factory.java
@@ -50,7 +50,7 @@
    * @return <code>true</code> if native-bzip2 is loaded & initialized 
    *         and can be loaded for this job, else <code>false</code>
    */
-  public static boolean isNativeBzip2Loaded(Configuration conf) {
+  public static synchronized boolean isNativeBzip2Loaded(Configuration conf) {
     String libname = conf.get("io.compression.codec.bzip2.library", 
                               "system-native");
     if (!bzip2LibraryName.equals(libname)) {
-- 
1.7.9.5

